### PR TITLE
Add WinHttp cmake option to the contributing guide

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/1300

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -143,6 +143,11 @@ The following CMake options are available for adding/removing project features.
 <td>OFF</td>
 </tr>
 <tr>
+<td>BUILD_TRANSPORT_WINHTTP</td>
+<td>Build the WinHTTP transport adapter. When building on Windows systems, if no other transport adapter is built, this option will be automatically turned ON. This option is not supported on non-Windows OSes.</td>
```
Comment (RickWinter) [inline CONTRIBUTING.md:147]:
```suggestion
<td>Build the WinHTTP transport adapter. When building on Windows systems, if no other transport adapter is built, this option is ON by default. This option is not supported on non-Windows OSes.</td>
```

**Review by RickWinter — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (RickWinter) [reply CONTRIBUTING.md:147]: Its a NIT, Up to you if you want to use the suggested wording.

Comment (ahsonkhan) [reply CONTRIBUTING.md:147]: The option isn't actually ON by default. It is turned on by one of our cmake modues, only under certain conditions.
This is similar wording to the curl option above.
https://github.com/Azure/azure-sdk-for-cpp/blob/9ed77d8b0648588432ca3e3c26db911c9041c396/CMakeLists.txt#L13
https://github.com/Azure/azure-sdk-for-cpp/blob/9ed77d8b0648588432ca3e3c26db911c9041c396/cmake-modules/AzureTransportAdapters.cmake#L25-L27

**Pull request merged**